content/docs/extension-author-guide/create-extension/run-dockerfile.md

```diff
@@ -0,0 +1,111 @@
++++
+title="Generating a run.Dockerfile"
+weight=405
++++
+
+### Examine `curl` extension
+
+#### detect
+
+`cat $workspace/samples/extensions/curl/bin/detect` - the extension always detects and provides a dependency
+  called `curl`
+
+#### generate
+
+`cat $workspace/samples/extensions/curl/bin/generate` - the extension generates a `run.Dockerfile` that switches the run image to
+  reference `run-image-curl`
+
+### Build a run image for `curl` extension to use
+
+1. `cat $workspace/samples/stacks/alpine/run/curl.Dockerfile` - this is a simple Dockerfile that creates a CNB run image
+  from the `curl` base image by adding the required CNB user configuration and `io.buildpacks.stack.id` label
+   * The Dockerfile could come from anywhere, but we include it in the `stacks` directory for convenience
+
+2. Build the image:
+
+```
+docker build \
+  --file $workspace/samples/stacks/alpine/run/curl.Dockerfile \
+  --tag run-image-curl .
+```
+
+### Re-create our builder with `hello-extensions` updated to require `curl`
+
+1. Edit `$workspace/samples/buildpacks/hello-extensions/bin/detect` to uncomment the second set of lines that
+  output `[[requires]]` to the build plan
+
+2. Create the builder:
+
+```
+pack builder create $registry_namespace/extensions-builder \
+  --config $workspace/samples/builders/alpine/builder.toml \
+  --publish
+```
+
+### Build the application image
+
+```
+pack build hello-extensions \
+  --builder $registry_namespace/extensions-builder \
+  --lifecycle-image $LIFECYCLE_IMAGE \
+  --verbose
+```
+
+You should see:
+
+```
+[detector] ======== Results ========
+[detector] pass: samples/tree@0.0.1
+[detector] pass: samples/curl@0.0.1
+[detector] pass: samples/hello-extensions@0.0.1
+[detector] Resolving plan... (try #1)
+[detector] samples/tree             0.0.1
+[detector] samples/curl             0.0.1
+[detector] samples/hello-extensions 0.0.1
+[detector] Running generate for extension samples/tree@0.0.1
+...
+[detector] Running generate for extension samples/curl@0.0.1
+...
+[detector] Checking for new run image
+[detector] Found a run.Dockerfile configuring image 'run-image-curl' from extension with id 'samples/curl'
+...
+[extender] Found build Dockerfile for extension 'samples/tree'
+[extender] Applying the Dockerfile at /layers/generated/build/samples_tree/Dockerfile...
+...
+[extender] Running build command
+[extender] ---> Hello Extensions Buildpack
+[extender] tree v1.8.0 (c) 1996 - 2018 by Steve Baker, Thomas Moore, Francesc Rocher, Florian Sesser, Kyosuke Tokoro
+...
+Successfully built image hello-extensions
+```
+
+### See the image run successfully
+
+`docker run hello-extensions`
+
+You should see something akin to:
+
+```
+curl 7.85.0-DEV (x86_64-pc-linux-musl)
```

Comment:
fwiw my full output at this step was:

```
curl 7.87.0-DEV (x86_64-pc-linux-musl) libcurl/7.87.0-DEV OpenSSL/1.1.1s zlib/1.2.12 brotli/1.0.9 libssh2/1.10.0 nghttp2/1.47.0
Release-Date: [unreleased]
Protocols: dict file ftp ftps gopher gophers http https imap imaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTPS-proxy IPv6 Largefile libz NTLM NTLM_WB SSL threadsafe TLS-SRP UnixSockets
```

I very briefly thought something had failed!   might be worth noting either ,  "the first line out the output should be akin to" ,  or putting some `...` after the same provided?

Reply:
Updated in 65f83e59